aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -1642,6 +1669,102 @@ static void apply_geqrf(Tensor& self, Tensor& tau, int64_t m, int64_t n) {
 #endif
 }
 
+static void geqrf_out_helper(const Tensor& input, const Tensor& QR, const Tensor& tau) {
+  TORCH_INTERNAL_ASSERT(input.dim() >= 2);
```

Comment:
Do you want to make the TORCH_INTERNAL_ASSERT_DEBUG_ONLY? You've already checked all these conditions, and this function is not user facing.